src/coreclr/jit/emit.cpp

```diff
@@ -4102,6 +4117,213 @@ void emitter::emitDispCommentForHandle(size_t handle, GenTreeFlags flag)
 #endif // DEBUG
 }
 
+//****************************************************************************
+// emitRemoveJumpToNextInst:  Checks all jumps in the jump list to see if they are
+//   unconditional jumps generated at the end of BB_ALWAYS basic blocks and whether
+//   they are now a jump to the next instruction, this can happen when the following
+//   basic block emitted no instructions
+//
+// Returns:
+//   true if any jumps have been removed, otherwise returns false
+//
+// Assumptions:
+//    the jump list must be ordered by increasing igNum+insNo
+//
+void emitter::emitRemoveJumpToNextInst()
+{
+#ifdef TARGET_XARCH
+    JITDUMP("*************** In emitRemoveJumpToNextInst()\n");
```

Comment:
When we're here, do we know if there are any candidates (so can short-circuit return if there aren't any)?

Reply:
No. We could have an emitter level field which keeps track of whether any candidate jumps have been added but given that 99% of compiled code seems to have some form of conditional jump that can be removed based on the spmi results it doesn't feel like it would save very much work.

Reply:
>  it doesn't feel like it would save very much work.

I disagree.

>  99% of compiled code seems to have some form of conditional jump that can be removed 

That's not true. If you just see the methods improved in the [diff](https://dev.azure.com/dnceng/public/_build/results?buildId=1787534&view=ms.vss-build-web.run-extensions-tab) for windows/x64:
- benchmarks collection: This PR improves 204 methods out of 37864 methods we have (0.5%) 
- coreclr_test collection: This PR improves 44821 methods out of 266018 methods we have (16%)

So for the remaining methods that are not optimized, we should have a way to short-circuit iterating the jump list.

Reply:
So do you want an emitter level field added?

Reply:
Ok, added. I called it `emitContainsCandidateJumpsToNextInst` I added an early out at the start of `emitRemoveJumpToNextInst` so if there are no candidates no work is done.